### PR TITLE
ASoC: SoundWire codecs: simplify/fix set_stream in codecs

### PR DESCRIPTION
**Pull request by plbossart**
We knew that the code could be simplified to avoid the dynamic allocation of a structure containing a single pointer, but in addition the tests with the kconfig/memory-debug-defconfig expose warnings reported by kmemleak.

Let's simplifiy all this.

### CODE REVIEW

**Comment by plbossart** [comment]
@shumingfan @oder-chiou @jack-cy-yu @rfvirgil @ryans-lee FYI

**Review by bardliao — APPROVED** [review]

File: sound/soc/codecs/sdw-mockup.c [hunk 1]
```diff
@@ -45,31 +41,15 @@ static const struct snd_soc_component_driver snd_soc_sdw_mockup_component = {
...
-
-	stream = kzalloc(sizeof(*stream), GFP_KERNEL);
-	if (!stream)
-		return -ENOMEM;
-
-	stream->sdw_stream = sdw_stream;
-
-	/* Use tx_mask or rx_mask to configure stream tag and set dma_data */
-	snd_soc_dai_dma_data_set(dai, direction, stream);
+	snd_soc_dai_dma_data_set(dai, direction, sdw_stream);
```
Comment (ujfalusi) [inline sound/soc/codecs/sdw-mockup.c:44]: wow, it did this, really?

**Review by ujfalusi — APPROVED** [review]
there must have been some idea (which got dropped) behind of the `struct sdw_stream_data` use to store a single pointer...

**Review by RanderWang — APPROVED** [review]
LGTM

File: sound/soc/codecs/sdw-mockup.c (hunk 1 above)
Comment (plbossart) [reply sound/soc/codecs/sdw-mockup.c:44]: @ujfalusi This was in the original code when I took over in 2018

````c
struct sdw_stream_data {
    struct sdw_stream_runtime *sdw_stream;
    int num_channels;
};

	if (stream->num_channels)
		num_channels = stream->num_channels;
	else
		num_channels = params_channels(params);

int rt700_channel_map(struct snd_soc_dai *dai,
		unsigned int tx_num, unsigned int *tx_slot,
		unsigned int rx_num, unsigned int *rx_slot)
{
	struct sdw_stream_data *dma;

	if (tx_num) {
		dma = dai->playback_dma_data;
		dma->num_channels = tx_num;
	} else {
		dma = dai->capture_dma_data;
		dma->num_channels = rx_num;
	}
````

but the set_channel_map and num_channels was removed, so the whole thing is useless now.

in hindsight, I also don't recall why we added a set_stream() API which is now limited to set_dmadata() now. Oh well.

**Pull request merged**